Gem/Code/Source/Components/NetworkCoinComponent.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Source/Components/NetworkCoinComponent.h>
+
+#include <AzCore/Serialization/SerializeContext.h>
+#include <Multiplayer/Components/NetworkTransformComponent.h>
+
+namespace MultiplayerSample
+{
+    void NetworkCoinComponent::Reflect(AZ::ReflectContext* context)
+    {
+        AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context);
+        if (serializeContext)
+        {
+            serializeContext->Class<NetworkCoinComponent, NetworkCoinComponentBase>()
+                ->Version(1);
+        }
+        NetworkCoinComponentBase::Reflect(context);
+    }
+
+    void NetworkCoinComponent::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+        if (IsNetEntityRoleClient())
+        {
+            m_rootLocation = GetEntity()->GetTransform()->GetWorldTranslation();
+            GetNetworkTransformComponent()->TranslationAddEvent(m_networkLocationHandler);
+
+            m_clientAnimationEvent.Enqueue(AZ::Time::ZeroTimeMs, true);
```

Comment:
Might be worth a comment that ZeroTime with repeating enabled uses the game tick rate; at first zero seems like repeated instantly.

Reply:
Sure.